app/forms/candidate_interface/equality_and_diversity/ethnic_group_form.rb

```diff
@@ -23,7 +23,11 @@ def save(application_form)
       else
         application_form.equality_and_diversity['ethnic_group'] = ethnic_group
 
-        if current_ethnic_group && ((ethnic_group == 'Prefer not to say') || (current_ethnic_group != ethnic_group))
+        if current_ethnic_group && ethnic_group == 'Prefer not to say'
```

Comment:
This will save the `hesa_etnicity` in the DB. Previously this was set to nil which we don't want when `Prefer not to say` is the option